src/T2DMap.cpp

```diff
@@ -3647,144 +3551,35 @@ void T2DMap::slot_setSymbol()
-                                                 "in the QComboBox it is put in.")
-                                              .arg(itSymbolUsed.key())
-                                              .arg(QString::number(itSymbolUsed.value())));
-                    }
-                }
-            }
-            newSymbol = QInputDialog::getItem(this,                    // QWidget * parent
-                                              tr("Enter room symbol",
-                                                 "Title for room symbol selection/entry dialog (used in three places)"),
-                                              tr("Choose:\n"
-                                                 " • an existing symbol from the list below (sorted by most commonly used first)\n"
-                                                 " • enter one or more graphemes (\"visible characters\") as a new symbol\n"
-                                                 " • enter a space to clear any existing symbols\n"
-                                                 "for all of the %n selected room(s):",
-                                                 // Intentional comment to separate arguments!
-                                                 "Use line feeds to format text into a reasonable rectangle if needed, "
-                                                 "%n is the number of rooms involved.",
-                                                 mMultiSelectionSet.size()), // const QString & label
-                                              displayStrings,                                                   // QStringList & items
-                                              0,                                                                // int current = 0
-                                              true,                                                             // bool editable = true
-                                              &isOk,                                                            // bool * ok = 0
-                                              Qt::Dialog);
-            if (isOk && displayStrings.contains(newSymbol)) {
-                // The user has selected one of the existing items in the form
-                // "XXXX {count:##}" and we need to chop off the stuff after the
-                // "XXXX" to get what is needed:
-
-                QRegularExpression countStripper(QStringLiteral("^(.*) {.*}$"));
-                QRegularExpressionMatch match = countStripper.match(newSymbol);
-                if (match.hasMatch() && match.lastCapturedIndex() > 0) {
-                    // captured(0) is the whole string that matched, which is
-                    // not what we want:
-                    newSymbol = match.captured(1);
-                }
-            }
-        }
+        QString inputSymbol = QString();
+        auto symbolDialog = new dlgRoomSymbol(mpHost, this);
```

Comment:
I can't figure out how custom lines dialog is done. It blocks map interactions, but does not block main window... This could be applied.
In theory it has `exec` call which make this window modal... but when I do set modality on symbol dialog it blocks main window as well.

Reply:
Remember the fix for multiple About buttons - same here, store a pointer and if the dialog is open, don't show another one.